srp/views.py

```diff
@@ -219,25 +224,35 @@ def srp_request_view(request, fleet_srp):
 
             try:
                 srp_kill_link = srpManager.get_kill_id(srp_request.killboard_link)
-                (ship_type_id, ship_value) = srpManager.get_kill_data(srp_kill_link)
+                (ship_type_id, ship_value, victim_name) = srpManager.get_kill_data(srp_kill_link)
             except ValueError:
                 logger.debug("User %s Submitted Invalid Killmail Link %s or server could not be reached" % (
                     request.user, srp_request.killboard_link))
                 # THIS SHOULD BE IN FORM VALIDATION
                 messages.error(request,
                                _("Your SRP request Killmail link is invalid. Please make sure you are using zKillboard."))
                 return redirect("auth_srp_management_view")
-            srp_ship_name = EveManager.get_itemtype(ship_type_id).name
-            srp_request.srp_ship_name = srp_ship_name
-            kb_total_loss = ship_value
-            srp_request.kb_total_loss = kb_total_loss
-            srp_request.post_time = post_time
-            srp_request.save()
-            completed = True
-            logger.info("Created SRP Request on behalf of user %s for fleet name %s" % (
-                request.user, srp_fleet_main.fleet_name))
-            messages.success(request, _('Submitted SRP request for your %(ship)s.') % {"ship": srp_ship_name})
 
+            characters = EveManager.get_characters_by_owner_id(request.user.id)
+            for character in characters:
+                if character.character_name == victim_name:
+                    srp_ship_name = EveManager.get_itemtype(ship_type_id).name
+                    srp_request.srp_ship_name = srp_ship_name
+                    kb_total_loss = ship_value
+                    srp_request.kb_total_loss = kb_total_loss
+                    srp_request.post_time = post_time
+                    srp_request.save()
+                    completed = True
+                    logger.info("Created SRP Request on behalf of user %s for fleet name %s" % (
+                        request.user, srp_fleet_main.fleet_name))
+                    messages.success(request, _('Submitted SRP request for your %(ship)s.') % {"ship": srp_ship_name})
+                    return redirect("auth_srp_management_view")
+                else:
+                    continue
+            messages.error(request,
+                           _("The character %(charname)s does not belong to your AllianceAuth account. Please add the API key for this character and try again") 
```

Comment:
Break the string up if you can, because its way over the 120 character line length.